src/source/JavaGenerator.scala

```diff
@@ -357,6 +357,131 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
         }
         w.wl
 
+        if (spec.javaImplementAndroidOsParcelable) {
+          // CREATOR
```

Comment:
Can you pull out these various long chunks of code into helper functions each with a clear purpose?